### PR TITLE
Describe the Interpolated line symbol layer

### LINKED ISSUE #6692: interpolated line symbol layer for vector layer (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#42676
Author: @vcloarec
QGIS version: 3.20

**interpolated line symbol layer for vector layer**

### PR Description:
This PR introduces a new ~~renderer~~ symbol layer for line vector layers. This symbol layer is the porting of the mesh layer's edge rendering.

interpolatedLine

The user can associate attributes or expressions to each line string's extremity (or curve). Value is interpolated along the line string between extremities.
The rendering is done with color and/or width 
varying depending on the interpolated value, resulting in a color and width varying line.
The color rendering is set with a color ramp shader widget (like raster and mesh layers).

Funded by Lutra Consulting

Note: text above was changed following the discussion below.

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #6692 

TODO: 
- [x] Add an illustration

### CODE REVIEW

File: docs/user_manual/style_library/symbol_selector.rst [hunk 1]
```diff
@@ -323,6 +323,26 @@ layer types:
...
+  geometry using feature attributes (or an expression) with a range of values.
+  Necessary inputs are:
+
+  * :guilabel:`Start value` and :guilabel:`End value`: XXX
+  * :guilabel:`Min. value` and :guilabel:`Max. value`: XXX
+    Press the |refresh| :sup:`Load` button to automatically fill them based
+    on start or end values expressions.
+  * Only available for the stroke option:
+
+    * :guilabel:`Min. width` and :guilabel:`Max. width`: define how wide
```
Comment (vcloarec) [inline docs/user_manual/style_library/symbol_selector.rst:339]:
```suggestion
    * :guilabel:`Min. width` and :guilabel:`Max. width`: define the range of the varying width. `Min. Width` corresponds to `Min. Value` and `Max. Width` corresponds to `Max. Value`
```

File: docs/user_manual/style_library/symbol_selector.rst [hunk 2]
```diff
@@ -323,6 +323,26 @@ layer types:
...
+      the line can be, linearly scaling the range of values onto XXX.
```
Comment (vcloarec) [inline docs/user_manual/style_library/symbol_selector.rst:340]:
```suggestion
```

File: docs/user_manual/style_library/symbol_selector.rst [hunk 3]
```diff
@@ -323,6 +323,26 @@ layer types:
...
+      A :ref:`unit <unit_selector>` can be associated.
+    * |checkbox| :guilabel:`Use absolute value`: XXX
```
Comment (vcloarec) [inline docs/user_manual/style_library/symbol_selector.rst:342]:
```suggestion
    * |checkbox| :guilabel:`Use absolute value`: only consider absolute value for interpolation (negative value are use as positive).
```

File: docs/user_manual/style_library/symbol_selector.rst [hunk 4]
```diff
@@ -323,6 +323,26 @@ layer types:
...
+    * |checkbox| :guilabel:`Ignore out of range`: XXX
```
Comment (vcloarec) [inline docs/user_manual/style_library/symbol_selector.rst:343]:
```suggestion
    * |checkbox| :guilabel:`Ignore out of range`: value out of the range `Min. Value`/`Max. Value` are not rendered.
```

File: docs/user_manual/style_library/symbol_selector.rst [hunk 5]
```diff
@@ -323,6 +323,26 @@ layer types:
   shapeburst to render the arrow body. Combined with the geometry generator,
   this type of layer symbol helps you representing flow maps.
 * **Geometry generator** (see :ref:`geometry_generator_symbol`)
+* **Interpolated line**: allows to render a line whose :guilabel:`Stroke width`
+  and/or :guilabel:`Color` may be constant (given a :guilabel:`Width` and
+  :guilabel:`Color` parameters) or vary along the road.
```
Comment (DelazJ) [inline docs/user_manual/style_library/symbol_selector.rst:328]:
```suggestion
  and/or :guilabel:`Color` may be constant (given a :guilabel:`Fixed width` and
  :guilabel:`Single color` parameters) or vary along the geometry.
```

File: docs/user_manual/style_library/symbol_selector.rst [hunk 6]
```diff
@@ -323,6 +323,26 @@ layer types:
...
+  When varying, the width and color of the line are interpolated along the
+  geometry using feature attributes (or an expression) with a range of values.
+  Necessary inputs are:
```
Comment (DelazJ) [inline docs/user_manual/style_library/symbol_selector.rst:331]:
```suggestion
  When varying, necessary inputs are:
```

**Comment by DelazJ** [comment]
> @DelazJ , sorry for the delay...

me feeling really ashamed for mine. Thanks for the feedback.

**Pull request merged**